src/main/java/uk/gov/hmcts/cmc/claimstore/events/ccd/CCDCaseHandler.java

```diff
@@ -148,6 +147,26 @@ public void linkDefendantToClaim(CCDLinkDefendantEvent event) {
         }
     }
 
+    @TransactionalEventListener
+    @LogExecutionTime
+    public void saveClaimDocument(CCDSaveClaimDocumentEvent event) {
+        try {
+            Claim ccdClaim = ccdCaseRepository.getClaimByExternalId(
+                event.getClaim().getExternalId(),
+                event.getAuthorisation()
+            ).orElseThrow(IllegalStateException::new);
+
+            ccdCaseRepository.saveClaimDocuments(event.getAuthorisation(),
+                ccdClaim.getId(),
+                event.getClaimDocumentCollection(),
+                event.getClaimDocumentType()
+            );
+        } catch (FeignException e) {
+            appInsights.trackEvent(CCD_ASYNC_FAILURE, REFERENCE_NUMBER, event.getClaim().getReferenceNumber());
```

Comment:
why are we catching only FeignException . what happens if there is any other exception.

Reply:
All the methods in CCDCaseHandler do that, so it follows the pattern. But yeah, it's a bit strange.

Reply:
This is strange. anyways its just a suggestion. This can be fixed , if need be, in a separate pr